src/tracer_options.cpp

```diff
@@ -105,8 +131,10 @@ ot::expected<std::set<PropagationStyle>> asPropagationStyle(
   return propagation_styles;
 }
 
-ot::expected<TracerOptions, const char *> applyTracerOptionsFromEnvironment(
+ot::expected<TracerOptions, std::string> applyTracerOptionsFromEnvironment(
     const TracerOptions &input) {
+  using namespace std::string_literals;
```

Comment:
I'd say the trailing `s` is both explicit and subtle.

We could adopt a house style that says "avoid use of special string literals, because they are easily confused with built-in literals and may surprise the reader."

I'd prefer to use the literals instead.